### ISSUE TITLE
All Insights except Trends bug out on refresh / direct link access

### ISSUE THREAD

**Issue by neilkakkar** [PostHog/posthog#8960]
## Bug description

*Please describe.*  
*If this affects the front-end, screenshots would be of great help.*  

![2022-03-10 11 44 33](https://user-images.githubusercontent.com/7115141/157663478-8a1b10fe-7e4a-42fc-b716-51c467985e76.gif)

## Expected behavior

There's no infinite spinner

## How to reproduce

1. Go to any insight except trends, wait for results to load
2. Refresh
3.

## Environment

- [ ] PostHog Cloud
- [ ] self-hosted PostHog (ClickHouse-based), version/commit: _please provide_
- [ ] self-hosted PostHog (Postgres-based, legacy), version/commit: _please provide_

## Additional context

I tried debugging this for a bit, [here's whats happening](https://posthog.slack.com/archives/C01G8S5T08Z/p1646913081983069).

To summarise, on refresh, there are 2 competing calls: `loadInsight` called by mount, and `setFilters` (which in turn calls `loadResults`) called by urlToAction. Usually, since results are already cached, results return first.

However, since the default insight is saved as a trend, the loadInsight returns a trend. This causes a mismatch between activeView and loadedView, creating an infinite spinner.
